src/preamble.js

```diff
@@ -1714,6 +1714,12 @@ function intArrayFromString(stringy, dontAddNull, length) {
 {{{ maybeExport('intArrayFromString') }}}
 
 function intArrayToString(array) {
+  if (typeof TextDecoder !== 'undefined') {
```

Comment:
Oh right, I wanted to ask about this. It was copied over from my `bytesToString` implementation and meant as just an optimization — confirmed that both `TextDecoder` without `fatal: true` and `Buffer` would return weird strings in the case of invalid input rather than throw exceptions — but as-is doesn't do anything assertion-related. I could disable the optimization when assertions are enabled, or if you think it makes more sense just remove that change entirely (and maybe submit it as a different PR).